bigframes/core/joins/name_resolution.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+
+from typing import Mapping, Sequence, Tuple
+
+
+class JoinNameRemapper:
+    def __init__(self, namespace: str) -> None:
+        self._namespace = namespace
+
+    def __call__(
+        self, left_column_ids: Sequence[str], right_column_ids: Sequence[str]
+    ) -> Tuple[Mapping[str, str], Mapping[str, str]]:
+        """When joining column ids from different namespaces, this function defines how names are remapped. Map only non-hidden ids."""
```

Comment:
I see here you warn to "Map only non-hidden IDs" but we are passing in some hidden columns in bigframes/core/joins/row_identity.py

Reply:
Adjusted code comment to reflect the real constraint: hidden ids should be mapped in a separate namespace as they are compile context-dependent, while value columns names must be fixed regardless of compilation modes.